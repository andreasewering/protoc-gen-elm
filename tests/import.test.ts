import {
  compileElm,
  getGeneratedFileContents,
  runPlugin,
} from "./snapshot_test_base";

describe("protoc-gen-elm", () => {
  beforeAll(() => runPlugin(["imported.proto", "importing.proto"]));
  const expectedElmFileNames = ["Proto/Imported.elm", "Proto/Importing.elm"];

  it("generates a valid elm file for imported.proto and importing.proto", async () => {
    await compileElm(expectedElmFileNames);
  });

  it("generates the expected code for imported.proto and importing.proto", async () => {
    const generatedContent = await getGeneratedFileContents(
      expectedElmFileNames
    );
    expect(generatedContent).toMatchSnapshot();
  });
});
