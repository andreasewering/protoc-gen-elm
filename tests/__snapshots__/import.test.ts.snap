// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protoc-gen-elm generates the expected code for imported.proto and importing.proto 1`] = `
"{- !!! DO NOT EDIT THIS FILE MANUALLY !!! -}


module Proto.Imported exposing
    ( Imported
    , importedDecoder
    , toImportedEncoder
    )

{-| ProtoBuf module: \`Proto.Imported\`

This module was generated automatically using

  - [\`protoc-gen-elm\`](https://www.npmjs.com/package/protoc-gen-elm) 1.0.0
  - \`protoc\` 3.14.0
  - the following specification file: \`imported.proto\`

To run it use [\`elm-protocol-buffers\`](https://package.elm-lang.org/packages/eriktim/elm-protocol-buffers/1.1.0) version 1.1.0 or higher.


# Model

@docs Imported


# Decoder

@docs importedDecoder


# Encoder

@docs toImportedEncoder

-}

import Protobuf.Decode as Decode
import Protobuf.Encode as Encode



-- MODEL


{-| \`Imported\` message
-}
type alias Imported =
    { first : String
    , second : Bool
    }



-- DECODER


{-| \`Imported\` decoder
-}
importedDecoder : Decode.Decoder Imported
importedDecoder =
    Decode.message (Imported \\"\\" False)
        [ Decode.optional 1 Decode.string setFirst
        , Decode.optional 2 Decode.bool setSecond
        ]



-- ENCODER


{-| \`Imported\` encoder
-}
toImportedEncoder : Imported -> Encode.Encoder
toImportedEncoder model =
    Encode.message
        [ ( 1, Encode.string model.first )
        , ( 2, Encode.bool model.second )
        ]



-- SETTERS


setFirst : a -> { b | first : a } -> { b | first : a }
setFirst value model =
    { model | first = value }


setSecond : a -> { b | second : a } -> { b | second : a }
setSecond value model =
    { model | second = value }

{- !!! DO NOT EDIT THIS FILE MANUALLY !!! -}


module Proto.Importing exposing
    ( Nested
    , nestedDecoder
    , toNestedEncoder
    )

{-| ProtoBuf module: \`Proto.Importing\`

This module was generated automatically using

  - [\`protoc-gen-elm\`](https://www.npmjs.com/package/protoc-gen-elm) 1.0.0
  - \`protoc\` 3.14.0
  - the following specification file: \`importing.proto\`

To run it use [\`elm-protocol-buffers\`](https://package.elm-lang.org/packages/eriktim/elm-protocol-buffers/1.1.0) version 1.1.0 or higher.


# Model

@docs Nested


# Decoder

@docs nestedDecoder


# Encoder

@docs toNestedEncoder

-}

import Proto.Imported exposing (..)
import Protobuf.Decode as Decode
import Protobuf.Encode as Encode



-- MODEL


{-| \`Nested\` message
-}
type alias Nested =
    { normalProperty : String
    , nestedProperty : Maybe Imported
    }



-- DECODER


{-| \`Nested\` decoder
-}
nestedDecoder : Decode.Decoder Nested
nestedDecoder =
    Decode.message (Nested \\"\\" Nothing)
        [ Decode.optional 1 Decode.string setNormalProperty
        , Decode.optional 2 (Decode.map Just importedDecoder) setNestedProperty
        ]



-- ENCODER


{-| \`Nested\` encoder
-}
toNestedEncoder : Nested -> Encode.Encoder
toNestedEncoder model =
    Encode.message
        [ ( 1, Encode.string model.normalProperty )
        , ( 2, (Maybe.withDefault Encode.none << Maybe.map toImportedEncoder) model.nestedProperty )
        ]



-- SETTERS


setNormalProperty : a -> { b | normalProperty : a } -> { b | normalProperty : a }
setNormalProperty value model =
    { model | normalProperty = value }


setNestedProperty : a -> { b | nestedProperty : a } -> { b | nestedProperty : a }
setNestedProperty value model =
    { model | nestedProperty = value }
"
`;
